### PR TITLE
revert: chore: deprecate client options presence

### PR DESCRIPTION
**Pull request by almeidx**
This reverts commit 8f97d2bacf78544e60935405316024a6bc4a4a50.

After internal discussion, we decided to revert this change. See https://github.com/discordjs/discord.js/pull/10426#discussion_r1703174908 for more info

### CODE REVIEW

**Review by didinele — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -6878,7 +6877,7 @@ export interface WebhookMessageCreateOptions extends Omit<MessageCreateOptions,
 export interface WebSocketOptions {
   large_threshold?: number;
   version?: number;
-  presence?: GatewayPresenceUpdateData;
+  presence?: PresenceData;
```
Comment (Qjuh) [inline packages/discord.js/typings/index.d.ts:6880]: Why was this changed? The goal here is to have those types being forward compatible and `@discordjs/ws` expects the previous type.

File: packages/discord.js/src/util/Options.js [hunk 2]
```diff
@@ -91,7 +91,7 @@ const { version } = require('../../package.json');
  * only set this if you know what you are doing</warn>
  * @property {BuildStrategyFunction} [buildStrategy] Builds the strategy to use for sharding
  * @property {IdentifyThrottlerFunction} [buildIdentifyThrottler] Builds the identify throttler to use for sharding
- * @property {PresenceData} [presence={}] Presence data to use upon login
+ * @property {PresenceData} [presence] Presence data to use upon login
```
Comment (Qjuh) [inline packages/discord.js/src/util/Options.js:94]: We should add `initialPresence` here because that’s what /ws uses and should be the property we suggest as alternative to the deprecated `ClientOptions#presence`.

File: packages/discord.js/src/client/Client.js [hunk 3]
```diff
@@ -230,6 +230,8 @@ class Client extends BaseClient {
       }
 
       this.options.ws.presence = this.presence._parse(this.options.presence);
+    } else if (this.options.ws.presence) {
+      this.options.ws.presence = this.presence._parse(this.options.ws.presence);
```
Comment (Qjuh) [inline packages/discord.js/src/client/Client.js:234]: This should parse `this.options.ws.initialPresence` with the ws.presence as fallback with nullish coalesce.

**Review by Qjuh — CHANGES_REQUESTED** [review]

**Review by vladfrangu — DISMISSED** [review]
Requesting changes too in case Qjuhs doesn't count

File: packages/discord.js/typings/index.d.ts (hunk 1 above)
Comment (almeidx) [reply packages/discord.js/typings/index.d.ts:6880]: I changed it to match the type in the JSDoc. I'll revert this and change it in the JSDoc instead

Comment (almeidx) [reply packages/discord.js/typings/index.d.ts]: Changing this to the gateway type forces users to pass all of its properties at all times, e.g.:
image

IMO it doesn't hurt to keep the parsing on our side... Thoughts? @discordjs/core

Comment (didinele) [reply packages/discord.js/typings/index.d.ts]: I'm fine with this as long as we do it exactly as we're doing intents (see Qjuh's /ws integration PR).

**Review by didinele — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]
